keyboards/gmmk/pro/ansi/keymaps/andrebrait/keymap.c

```diff
@@ -0,0 +1,143 @@
+    // Press Fn+N to toggle between 6KRO and NKRO. This setting is persisted to the EEPROM and thus persists between restarts.
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,           KC_MUTE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_HOME,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGDN,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT(
+        _______, KC_MYCM, KC_WHOM, KC_CALC, KC_MSEL, KC_MPRV, KC_MNXT, KC_MPLY, KC_MSTP, KC_MUTE, KC_VOLD, KC_VOLU, _______, KC_INS,           _______,
+        _______, RGB_TOG, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          KC_PSCR,
+        _______, _______, RGB_VAI, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET,            _______,
+        _______, _______, RGB_VAD, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, RGB_HUI, _______, _______, _______, NK_TOGG, _______, _______, _______, _______,          _______, RGB_MOD, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, RGB_SPD, RGB_RMOD, RGB_SPI
+    ),
+
+
+};
+
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+    return true;
+}
+
+#ifdef RGB_MATRIX_ENABLE
+void keyboard_post_init_user(void) {
+    /*
+    Custom handling of RGB_TOG means it is impossible to toggle the RGB Matrix on and off and we
+    need it to be enabled for the CAPS LOCK indicator to work, so enable it and then make sure it
+    gets persisted to the EEPROM.
+     */
+    if (!rgb_matrix_is_enabled()) {
+        rgb_matrix_enable_noeeprom();
+        eeconfig_update_rgb_matrix();
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case RGB_MOD:
+        case RGB_RMOD:
+        case RGB_HUI:
+        case RGB_HUD:
+        case RGB_SAI:
+        case RGB_SAD:
+        case RGB_VAI:
+        case RGB_VAD:
+        case RGB_SPI:
+        case RGB_SPD:
```

Comment:
> I guess it would pollute the discussion about this PR in specific. You can email me if you'd like, but I usually answer whatever questions people might have. It's informative, after all.
> 
> So, I assume you're also new to C or C-style languages like Java, right?
> The cases in a `switch` statement in C are _fall-through_. That means, once something matches, all lines of code below will be executed until there is a `break` or the function returns.
> 
> So that means a match for `RGB_VAI` would execute all lines after that `case` until the `break;` you can find in line 103.
> That `break;` is what prevents the execution to go further and causes the code to exit the `switch` statement.
> 
> This is a powerful thing. For cases like this, it saves us from doing something like:
> 
> ```c
> if (keycode == RGB_MOD
>     || keycode == RGB_RMOD
>     || keycode == RGB_HUI
>     ...
>     || keycode == RGB_SPD)
> {
>     ...
> } else if (keycode == RGB_TOG) 
> {
>     ...
> }
> ```
> 
> However, as you can probably imagine, you have to be careful when doing it. If I was to forget that `break;`, then a press on `RGB_HUI` would turn off my LEDs by executing the code that's in `case RGB_TOG` because of the fall-through.

Thank you - I completely missed the fall-through - certainly not 'new' to c but very rusty coming back to it for QMK (Had C style classes in the mid 90's CS major) - Deleted my comment to not clutter this page (No concerns w/you doing the same).